models/panic.py

```diff
@@ -56,26 +53,35 @@ def agent_action(agent, **kwargs):
             if DEBUG.debug:
                 print("Changing the agent's group to calm!")
             agent.has_acted = True
-            get_model(agent.exec_key).add_switch(str(agent), PANIC, CALM)
+            mdl.add_switch(str(agent), PANIC, CALM)
     return DONT_MOVE
 
 
-def start_panic(exec_key):
-    """
-    This function should be rewritten.
-    We will make a new group method called `get_rand_subset(n)`.
-    Then we will flip those agents to panicked.
-    """
-    maxPosn = panic_grps[CALM][WIDTH] * panic_grps[CALM][HEIGHT]
-    num_panic = panic_grps[PANIC][PANICKED]
```

Comment:
@gcallah 
Panic seems to be running okay now. I think the problem was trying to randomly get an agent on the board.
maxPosn is width * height, so randomly picking a number between 0 and maxPosn will be out of bounds. For example with height 10, 10, there wont be an agent Calm100 in the registry. 
This would trigger a load from the disk since registry assumes if it's not in the thread it must be on the disk.

Reply:
Ah, OK, that makes sense.